SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -744,16 +782,16 @@ void CSolver::InitiatePeriodicComms(CGeometry *geometry,
 
             /*--- Get conservative solution and rotate if necessary. ---*/
 
-            for (iVar = 0; iVar < nVar; iVar++)
-              rotPrim_i[iVar] = base_nodes->GetSolution(iPoint, iVar);
+            for (iVar = 0; iVar < ICOUNT; iVar++)
+              rotPrim_i[iVar] = field(iPoint, iVar);
 
             if (rotate_periodic) {
-              Rotate(zeros, &base_nodes->GetSolution(iPoint)[1], &rotPrim_i[1]);
+              Rotate(zeros, &field(iPoint,1), &rotPrim_i[1]);
```

Comment:
Periodic comms with rotation will not work for NEMO.
This velocity offset should be generalized.